src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```
@@ -578,12 +575,10 @@ public FormValidation doLoginCheck(@QueryParameter String name, @QueryParameter
             }
 
             final String defaultRegion = ClientHelper.DEFAULT_AMAZON_S3_REGION_NAME;
-            final AmazonS3 client = ClientHelper.createClient(
-                    checkedAccessKey, checkedSecretKey, useRole, defaultRegion, Jenkins.get().proxy);
 
-            try {
+            try (var client = ClientHelper.createClient(checkedAccessKey, checkedSecretKey, useRole, defaultRegion, Jenkins.get().getProxy())) {
                 client.listBuckets();
-            } catch (AmazonClientException e) {
+            } catch (Exception e) {
```

It will hide other error scenarios. Better use `software.amazon.awssdk.core.exception.SdkException` 
[Ref](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-exception-changes.html)